tiledb/sm/subarray/range_subset.cc

```diff
@@ -120,6 +120,23 @@ Status RangeSetAndSuperset::sort_ranges(ThreadPool* const compute_tp) {
   return impl_->sort_ranges(compute_tp, ranges_);
 }
 
+tuple<Status, optional<std::string>> RangeSetAndSuperset::add_range(
+    Range& range, const bool read_range_oob_error) {
+  // Check range is valid
+  impl_->check_range_is_valid(range);
```

Comment:
Did you mean to return `Status` here?

Reply:
No, it throws exceptions instead of returning a Status.